File: __old/abc309/src/bin/b.py
import copy
N = int(input())

A = []

for i in range(N):
	A.append(list(input()))


_A = copy.copy(A)

for i in range(N - 1):
	print(A[i][i + 1], A[0][i])
	_A[0][i + 1] = A[0][i]

# for i in range(N - 1):
# 	_A[i + 1][N - 1] = A[i][N - 1]

# for i in range(1, N):
# 	_A[N - 1][i - 1] = A[N - 1][i]

# for i in range(1, N):
# 	_A[i - 1][0] = A[i][0]

for i in range(N):
	for j in range(N):
		print(_A[i][j], end = '')
	print()

File: __old/abc309/src/bin/b.rs
use proconio::{fastout, input};

#[fastout]
fn main() {
    input! {
        _n: usize,
        _a: [String; _n],
    }

    let mut _newa: Vec<Vec<char>> = Vec::new();
    let mut _aa: Vec<Vec<char>> = Vec::new();

    let _nn = _n - 1;

    for _i in 0.._n {
        _newa.push(_a[_i].chars().collect());
        _aa.push(_a[_i].chars().collect());
    }

    for _i in 0.._nn {
        _newa[0][_i + 1] = _aa[0][_i];
    }

    for _i in 0.._nn {
        _newa[_i + 1][_nn] = _aa[_i][_nn];
    }

    for _i in 0.._nn {
        _newa[_nn][_nn - _i - 1] = _aa[_nn][_nn - _i];
    }

    for _i in 0.._nn {
        _newa[_nn - _i - 1][0] = _aa[_nn - _i][0];
    }

    for _i in 0.._n {
        println!("{}", _newa[_i].iter().collect::<String>());
    }
}
